src/delphi/dataset/token_map.py

```diff
@@ -0,0 +1,25 @@
+from typing import cast
+
+from datasets import Dataset
+
+
+def token_map(
+    tokenized_dataset: Dataset,
+) -> dict[int, list[tuple[int, int]]]:
+    """Return a mapping of tokens to their (prompt_idx, token_idx) locations in the tokenized_dataset.
+
+    Args:
+        tokenized_dataset (Dataset): A tokenized dataset.
+
+    Returns:
+        dict[int, list[tuple[int, int]]]: A mapping of tokens to their (prompt_idx, token_idx)
+            locations in the tokenized_dataset.
+    """
+    mapping = {}
+    tokenized_dataset = cast(Dataset, tokenized_dataset)
+    for prompt_idx, prompt in enumerate(tokenized_dataset):
+        prompt = cast(dict, prompt)
```

Comment:
what's the error here, pyright doesn't realize prompt has a __get_item__?

Reply:
Yes. this is the full message:
```
Argument of type "Literal['tokens']" cannot be assigned to parameter "__s" of type "slice" in function "__getitem__"
  "Literal['tokens']" is incompatible with "slice"
```
I tested the dataset live, and the prompt seems to be of type dict when I access it, so I casted it to dict. Should I get a typeignore comment in or do something else?